### ISSUE TITLE
Linchpin execution fails with Error executing hook: 'can't set attribute

### ISSUE THREAD

**Issue by mcornea** [CentOS-PaaS-SIG/linchpin#1134]
```shell
(rhhi-venv-TodCbu3) [root@sealusa35 linchpin_workspace]# linchpin --pinfile parsedPinFile --template-data @hooks/ansible/rhhi-setup/extravars.yaml -v destroy
/home/rhhi-ci/jenkins/workspace/rhhi.next-virt-customized/rhhi-venv-TodCbu3/lib/python2.7/site-packages/linchpin/utils/dataparser.py:99: YAMLLoadWarning: calling yaml.load() without Loader=... is deprecated, as the default Loader is unsafe. Please read https://msg.pyyaml.org/load for full details.
  data = yaml.load(data)
-------
start hook ansible:rhhi-setup
Error executing hook: 'can't set attribute'
```

The issue is fixed after downgrading ansible:

```shell
(rhhi-venv-TodCbu3) [root@sealusa35 linchpin_workspace]# pip install ansible==2.7.1
DEPRECATION: Python 2.7 will reach the end of its life on January 1st, 2020. Please upgrade your Python as Python 2.7 won't be maintained after that date. A future version of pip will drop support for Python 2.7.
Collecting ansible==2.7.1
Requirement already satisfied: setuptools in /home/rhhi-ci/jenkins/workspace/rhhi.next-virt-customized/rhhi-venv-TodCbu3/lib/python2.7/site-packages (from ansible==2.7.1) (41.0.1)
Requirement already satisfied: PyYAML in /home/rhhi-ci/jenkins/workspace/rhhi.next-virt-customized/rhhi-venv-TodCbu3/lib/python2.7/site-packages (from ansible==2.7.1) (5.1)
Requirement already satisfied: jinja2 in /home/rhhi-ci/jenkins/workspace/rhhi.next-virt-customized/rhhi-venv-TodCbu3/lib/python2.7/site-packages (from ansible==2.7.1) (2.10.1)
Collecting paramiko (from ansible==2.7.1)
  Using cached https://files.pythonhosted.org/packages/cf/ae/94e70d49044ccc234bfdba20114fa947d7ba6eb68a2e452d89b920e62227/paramiko-2.4.2-py2.py3-none-any.whl
Requirement already satisfied: cryptography in /home/rhhi-ci/jenkins/workspace/rhhi.next-virt-customized/rhhi-venv-TodCbu3/lib/python2.7/site-packages (from ansible==2.7.1) (2.6.1)
Requirement already satisfied: MarkupSafe>=0.23 in /home/rhhi-ci/jenkins/workspace/rhhi.next-virt-customized/rhhi-venv-TodCbu3/lib/python2.7/site-packages (from jinja2->ansible==2.7.1) (1.1.1)
Collecting pynacl>=1.0.1 (from paramiko->ansible==2.7.1)
  Using cached https://files.pythonhosted.org/packages/b3/25/e605574f24948a8a53b497744e93f061eb1dbe7c44b6465fc1c172d591aa/PyNaCl-1.3.0-cp27-cp27mu-manylinux1_x86_64.whl
Requirement already satisfied: pyasn1>=0.1.7 in /home/rhhi-ci/jenkins/workspace/rhhi.next-virt-customized/rhhi-venv-TodCbu3/lib/python2.7/site-packages (from paramiko->ansible==2.7.1) (0.4.5)
Collecting bcrypt>=3.1.3 (from paramiko->ansible==2.7.1)
  Using cached https://files.pythonhosted.org/packages/50/d8/95f7cb04344033bf9d1a12c5a7969a15999b6a710fbe1969c517333d9a62/bcrypt-3.1.6-cp27-cp27mu-manylinux1_x86_64.whl
Requirement already satisfied: enum34; python_version < "3" in /home/rhhi-ci/jenkins/workspace/rhhi.next-virt-customized/rhhi-venv-TodCbu3/lib/python2.7/site-packages (from cryptography->ansible==2.7.1) (1.1.6)
Requirement already satisfied: asn1crypto>=0.21.0 in /home/rhhi-ci/jenkins/workspace/rhhi.next-virt-customized/rhhi-venv-TodCbu3/lib/python2.7/site-packages (from cryptography->ansible==2.7.1) (0.24.0)
Requirement already satisfied: cffi!=1.11.3,>=1.8 in /home/rhhi-ci/jenkins/workspace/rhhi.next-virt-customized/rhhi-venv-TodCbu3/lib/python2.7/site-packages (from cryptography->ansible==2.7.1) (1.12.3)
Requirement already satisfied: six>=1.4.1 in /home/rhhi-ci/jenkins/workspace/rhhi.next-virt-customized/rhhi-venv-TodCbu3/lib/python2.7/site-packages (from cryptography->ansible==2.7.1) (1.12.0)
Requirement already satisfied: ipaddress; python_version < "3" in /home/rhhi-ci/jenkins/workspace/rhhi.next-virt-customized/rhhi-venv-TodCbu3/lib/python2.7/site-packages (from cryptography->ansible==2.7.1) (1.0.22)
Requirement already satisfied: pycparser in /home/rhhi-ci/jenkins/workspace/rhhi.next-virt-customized/rhhi-venv-TodCbu3/lib/python2.7/site-packages (from cffi!=1.11.3,>=1.8->cryptography->ansible==2.7.1) (2.19)
Installing collected packages: pynacl, bcrypt, paramiko, ansible
  Found existing installation: ansible 2.8.0
    Uninstalling ansible-2.8.0:
      Successfully uninstalled ansible-2.8.0
Successfully installed ansible-2.7.1 bcrypt-3.1.6 paramiko-2.4.2 pynacl-1.3.0
(rhhi-venv-TodCbu3) [root@sealusa35 linchpin_workspace]# linchpin --pinfile parsedPinFile --template-data @hooks/ansible/rhhi-setup/extravars.yaml -v destroy
/home/rhhi-ci/jenkins/workspace/rhhi.next-virt-customized/rhhi-venv-TodCbu3/lib/python2.7/site-packages/linchpin/utils/dataparser.py:99: YAMLLoadWarning: calling yaml.load() without Loader=... is deprecated, as the default Loader is unsafe. Please read https://msg.pyyaml.org/load for full details.
  data = yaml.load(data)
-------
start hook ansible:rhhi-setup
 [WARNING]: Unable to parse /home/rhhi-ci/jenkins/workspace/rhhi.next-virt-customized/rhhi-qe-core/s2i/linchpin-provisioner/linchpin_workspace/inventories/libvirt_network.inventory as an inventory source

 [WARNING]: No inventory was parsed, only implicit localhost is available


PLAY [Install requirements] ******************************************************************************************************************************************************************************************************************

TASK [Gathering Facts] ***********************************************************************************************************************************************************************************************************************
ok: [localhost]
```

PinFile:
```yaml
---
# If provisionhost_hostname is not set then assume it's a virtual environment
libvirt-network:
  topology:
    topology_name: "libvirt_network"
    resource_groups:
      - resource_group_name: baremetal_net
        resource_group_type: libvirt
        resource_definitions:
          - name: baremetal
            role: libvirt_network
            uri: qemu:///system
            forward_mode: nat
            ip: 192.168.123.1
            dhcp_start: 192.168.123.100
            dhcp_end: 192.168.123.150
            delete_on_destroy: yes
            domain: ostest.test.metalkube.org
            bridge: baremetal
            dns:
              forwarders:
                - domain: apps.ostest.test.metalkube.org
                  addr: 127.0.0.1
              hosts:
                - ip: 192.168.123.5
                  hostnames:
                    - api
                    - api-int
                - ip: 192.168.123.6
                  hostnames:
                    - ns1

      - resource_group_name: provisioning_net
        resource_group_type: libvirt
        resource_definitions:
          - name: provisioning
            role: libvirt_network
            uri: qemu:///system
            bridge: provisioning
            delete_on_destroy: yes

  hooks:
    predestroy:
      - name: rhhi-setup
        type: ansible
        context: True
        actions:
          - playbook: requirements.yaml

cfgs:
  libvirt:
    __IP__: name
    __ADDRESS__: ip

libvirt-new:
  topology:
    topology_name: rhhi
    resource_groups:
      - resource_group_name: rhhi-node
        resource_group_type: libvirt
        resource_definitions:
          - role: libvirt_node
            name: rhhi-node
            uri: qemu:///system
            count: 6
            image_src: http://cloud.centos.org/centos/7/images/CentOS-7-x86_64-GenericCloud.qcow2
            memory: 16384
            vcpus: 8
            arch: x86_64
            ssh_key: id_rsa
            additional_storage: 42G
            disk_type: virtio_scsi
            name_separator: '-'
            networks:
              - name: baremetal
              - name: provisioning
                dhcp: false
            storage:
              - name: osd0
                size: 8
                cache: unsafe
                units: G
                disk_type: virtio_scsi
                device: sdb
              - name: osd1
                size: 8
                cache: unsafe
                units: G
                disk_type: virtio_scsi
                device: sdc
              - name: osd2
                size: 8
                cache: unsafe
                units: G
                disk_type: virtio_scsi
                device: sdd

      - resource_group_name: provisionhost
        resource_group_type: libvirt
        resource_definitions:
          - role: libvirt_node
            name: provisionhost
            uri: qemu:///system
            count: 1
            image_src: http://cloud.centos.org/centos/7/images/CentOS-7-x86_64-GenericCloud.qcow2
            memory: 24576
            vcpus: 8
            cpu_mode: host-passthrough
            arch: x86_64
            additional_storage: 50G
            name_separator: '-'
            ssh_key: id_rsa
            networks:
              - name: baremetal
              - name: provisioning
                dhcp: false

  layout:
    inventory_layout:
      vars:
        hostname: __IP__
        ansible_ssh_host: __ADDRESS__
        ansible_ssh_user: centos
        ansible_ssh_common_args: '"-o StrictHostKeyChecking=no"'
      hosts:
        openshift:
          count: 6
          host_groups:
            - openshift
        provisionhost:
          host_groups:
            - provisionhost

  hooks:
    postdestroy:
      - name: rhhi-setup
        type: ansible
        context: True
        actions:
          - playbook: vbmc.yaml
            extra_vars: { "action": "cleanup" }
    postup:
      - name: rhhi-setup
        type: ansible
        context: True
        actions:
          - playbook: update-kernel.yaml
          - playbook: dns_update.yaml
            vars: extravars.yaml
          - playbook: nic_adjust.yaml
            vars: extravars.yaml
          - playbook: client_ssh_key.yaml
          - playbook: vbmc.yaml
            extra_vars: { "action": "install" }

```

**Comment by 14rcole** [comment]
Which version of ansible were you originally using?

**Comment by mcornea** [comment]
> Which version of ansible were you originally using?

ansible 2.8 which gets pulled via pip starting last week

**Comment by samvarankashyap** [comment]
should be fixed by #1154

**Comment by 14rcole** [comment]
fixed by #1154
